crates/dropshot-api-manager/src/cmd/diff.rs

```diff
@@ -0,0 +1,252 @@
+// Copyright 2025 Oxide Computer Company
+
+use crate::{
+    apis::ManagedApis,
+    environment::{BlessedSource, ResolvedEnv},
+    output::{OutputOpts, Styles, display_load_problems, write_diff},
+    spec_files_blessed::BlessedApiSpecFile,
+    spec_files_local::LocalApiSpecFile,
+};
+use anyhow::{Context, bail};
+use camino::Utf8Path;
+use dropshot_api_manager_types::ApiIdent;
+use owo_colors::OwoColorize;
+use similar::TextDiff;
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    io::Write,
+};
+
+/// Compare local OpenAPI documents against blessed (upstream) versions.
+///
+/// For each API with differences, shows the diff between what's on disk locally
+/// and what's blessed in the upstream branch (typically origin/main).
+///
+/// Diff output is written directly to `writer`. Headers describing each changed
+/// file are written to stderr.
+pub(crate) fn diff_impl<W: Write>(
+    apis: &ManagedApis,
+    env: &ResolvedEnv,
+    blessed_source: &BlessedSource,
+    output: &OutputOpts,
+    writer: &mut W,
+) -> anyhow::Result<()> {
+    let mut styles = Styles::default();
+    if output.use_color(supports_color::Stream::Stdout) {
+        styles.colorize();
+    }
```

Comment:
Is this correct, given that a writer is being passed in? We should probably have the writer indicate whether it's stdout or being captured (in which case don't colorize)